client/serviceinterface_update.go

```diff
@@ -329,7 +329,11 @@ func removeServiceInterfaceTarget(serviceName string, targetName string, deleteI
 			modified := false
 			targets := []types.ServiceInterfaceTarget{}
 			for _, t := range service.Targets {
-				if (t.Name == targetName || (t.Name == "" && targetName == serviceName)) && (t.Namespace == "" || t.Namespace == namespace) {
+				targetFound := t.Name == targetName
+				unqualifiedServiceFound := t.Name == "" && targetName == serviceName && t.Service == targetName
```

Comment:
Can you elaborate on the bug this is fixing? What were the args for the expose and the unexpose and what is the actual result v expected result?

Reply:
Actually the `unqualifiedServiceFound` (case) is not related to the issue itself.
It does not seem to be a valid condition to me anymore (or maybe/probably something I am not considering).

As is, meaning, without the latest `t.Service == targetName` comparison, it basically checks if the respective
target.Name is empty, then no other param for the respective target being iterated is validated, it simply 
compares the provided target name and the provided service address. This way if multiple targets have empty
names (which I believe is no longer a valid condition?), all of them would be removed.

Do you know/remember of something I am not considering for this case?
If not, do you think it should be removed?

Reply:
Can you give an example of a failing interaction?

Reply:
Sure. The latest case: `qualifiedServiceFound := t.Service == fmt.Sprintf("%s.%s", targetName, namespace)` is
actually fixing the related issue.

If you use a latest version of Skupper (1.4.2) and expose an existing k8s service, i.e.:

```
skupper expose service nginx --address nginx-skupper
```

then looking at the targets in the skupper-services ConfigMap, we have this:

```
"targets":[{"name":"nginx.sk1","service":"nginx.sk1","namespace":"sk1"}]
```

As you can see, the exposed k8s service is now qualified with the "target-namespace" option,
which was introduced recently (but it was not actually tied to the unexposed command before this PR).

Reply:
To me it is the behaviour for expose that seems wrong. If you don't specify a target-namespace for the service then the service as recorded in the target should be exactly what the user specified. The '.sk1' suffix seems wrong to me.

Reply:
I believe you're correct. Possibly it was done like this so that the connector is created using the value
stored in the "target.service" as the target host. I will modify the expose to stop storing the namespace as part
of the service name.

Now two questions.

1. Do you think it is worth adding an skupper update verification on eventual service
definitions with targets containing service names with the namespace suffix and fix that as well?

2. What do you think of the updated condition above (`unqualifiedServiceFound`)? Should it be kept in this PR?

Reply:
Answering my own questions. I believe that for the former 1st question, it is not needed to due a minor
impact, as it would only impact exposed services using the CLI (services exposed using annotations
should not be impacted).

For the 2nd question, I believe that the udpated logic seems right. So I think it should be kept.

Reply:
Generally, I believe what we want is that the arguments needed for a successful unexpose  should have values that match those provided for those same options on the expose.

In my opinion, the simplest solution is to state that the target namespace is not applicable when the target type is 'service'. It is possible to target a service in a different namespace simply by qualifying the service name provided. With that view we would check if target-namespace was specified for target type service and report an error if so.  Then unexpose only needs to test either the target-name+target-namespace or the servicename (which should be provided in the exact form used for expose).

To me this avoids unnecessary complication.

However if that approach is not desired, I think it would be better to ensure that the namespace of a target is only populated if the target-namespace option was specified. I.e. if it is the current namespace, it would be empty.

Reply:
> Generally, I believe what we want is that the arguments needed for a successful unexpose should have values that match those provided for those same options on the expose.

Agreed. I believe with the latest changes, this is how it is supposed to work now for services as well.

> In my opinion, the simplest solution is to state that the target namespace is not applicable when the target type is 'service'. It is possible to target a service in a different namespace simply by qualifying the service name provided. With that view we would check if target-namespace was specified for target type service and report an error if so. Then unexpose only needs to test either the target-name+target-namespace or the servicename (which should be provided in the exact form used for expose).

I have added one minor change that basically allows you to use it either way. So basically you can expose and unexpose as follows:

```
skupper expose service my-service.target-namespace --address my-skupper-service
skupper unexpose service my-service.target-namespace --address my-skupper-service
```

or

```
skupper expose service my-service --target-namespace target-namespace --address my-skupper-service
skupper unexpose service my-service --target-namespace target-namespace --address my-skupper-service
```

I believe this is a more uniform way to tell the story of exposing resources from different namespaces, as
services can be exposed as you mentioned (just by adding the . namespace) or using --target-namespace,
which makes it similar to how other kind of resources are exposed.

Reply:
I think the apparent uniformity is something of an illusion. For deployments etc, the target namespace is used to query the api server, which is why using it with those target types requires extra permissions. The way services are handled is very different. (Arguably the target type 'service' is misleading name)

With your changes `skupper expose service foo.bar --target-namespace baz --port 80 --address foo` would cause the baz to be silently ignored.  `skupper expose service east.database.acme.org --port 80 --address db` would be stored as service='east' namespace='database.acme.org' which is not ideal since the latter is not actually a valid namespace.

If we must keep the --target-namespace for services I think we should aim to have the target fields match what was actually specified by the user, without modification, or else we will not necessarily be able to determine that again should there be a reason to do so. I.e. the service field should always hold what was in the <target-name> argument and the namespace field what was in the --target-namespace option.

I increasingly feel that the best option would be to just say that the target-namespace is not compatible with the service target-type (since the mechanism that option actually is designed to invoke is not relevant for the way we handle service targets)

Reply:
I agree that the mixed behavior of exposing/unexposing a qualified service along with using the
--target-namespace flag is obscure and should be rejected.

This behavior is already present today in the expose command, this latest change just added to the
unexpose. So I think this PR should also remove that behavior from the expose command.

> The way services are handled is very different. (Arguably the target type 'service' is misleading name)

True. One interesting thing is that during expose, the target service is being verified in order to determine
the target ports. But when we do a service bind, then the target service name is handled merely as a hostname.
Using 'service' as the target type is really misleading in this case.

> I increasingly feel that the best option would be to just say that the target-namespace is not compatible with the service target-type (since the mechanism that option actually is designed to invoke is not relevant for the way we handle service targets)

I was leaning toward leaving both and reject mixing them by verifying if target service was qualified, but it might
be simpler to simply reject using the --target-namespace for target type == service and do as you suggested.

Once the definition monitor is updated to handle target service names from other namespaces, it will make 
usage seamless between CLI and the declarative approach as well.